uniflow-androidx/src/main/java/io/uniflow/androidx/flow/AndroidDataFlow.kt

```diff
@@ -44,25 +47,23 @@ abstract class AndroidDataFlow(
         defaultState: UIState = UIState.Empty,
         defaultCapacity: Int = Channel.BUFFERED,
         defaultDispatcher: CoroutineDispatcher = UniFlowDispatcher.dispatcher.io()
-) : ViewModel(), DataFlow {
-    private val tag = this.toString()
-    val dataPublisher: LiveDataPublisher = LiveDataPublisher(defaultState)
+) : ViewModel(), DataFlow, DataPublisher {
+    override val tag = this.toString()
+
     private val coroutineScope: CoroutineScope = viewModelScope
-    private val dataStore: UIDataStore = UIDataStore(dataPublisher, defaultState,tag)
-    private val reducer: ActionReducer = ActionReducer(dataStore, coroutineScope, defaultDispatcher, defaultCapacity,tag)
-    private val actionDispatcher: ActionDispatcher
-        get() = ActionDispatcher(viewModelScope, reducer, dataStore, this,tag)
 
-    final override fun getCurrentState() = actionDispatcher.getCurrentState()
-    final override fun <T : UIState> getCurrentStateOrNull(stateClass: KClass<T>): T? = actionDispatcher.getCurrentStateOrNull()
-    final override fun action(onAction: ActionFunction<UIState>): ActionFlow = actionDispatcher.action(onAction)
-    final override fun action(onAction: ActionFunction<UIState>, onError: ActionErrorFunction): ActionFlow = actionDispatcher.action(onAction, onError)
-    final override fun <T : UIState> actionOn(stateClass: KClass<T>, onAction: ActionFunction<T>): ActionFlow = actionDispatcher.actionOn(stateClass, onAction)
-    final override fun <T : UIState> actionOn(stateClass: KClass<T>, onAction: ActionFunction<T>, onError: ActionErrorFunction): ActionFlow = actionDispatcher.actionOn(stateClass, onAction, onError)
+    val defaultDataPublisher = liveDataPublisher(defaultState, "main")
+    override suspend fun publishState(state: UIState, pushStateUpdate: Boolean) = defaultPublisher().publishState(state, pushStateUpdate)
+    override suspend fun publishEvent(event: UIEvent) = defaultPublisher().publishEvent(event)
+    override suspend fun getState(): UIState = defaultPublisher().getState()
+    override fun defaultPublisher(): DataPublisher = defaultDataPublisher
```

Comment:
Same here why we are adding `default` at the beginning ?, do we have more options of `DataPublisher` ?? 
maybe we can just use name `publisher`